src/type.h

```diff
@@ -626,6 +626,8 @@ class ArrayType : public SequentialType {
                             example).
      */
     ArrayType(const Type *elementType, int numElements);
+    ArrayType(const Type *elementType, Symbol *num);
```

Comment:
The different type of element count argument probably requires some comments about their intended usage scenarios and meaning.

Reply:
Done